app/scripts/views/settings/communication_preferences.js

```diff
@@ -39,22 +39,35 @@ function (Cocktail, Xss, Constants, MarketingEmailErrors, BaseView, FormView,
     beforeRender: function () {
       var self = this;
       var emailPrefs = self.getMarketingEmailPrefs();
-
       return emailPrefs.fetch()
         .fail(function (err) {
           if (MarketingEmailErrors.is(err, 'UNKNOWN_EMAIL')) {
             // user has not yet opted in to Basket yet. Ignore.
             return;
           }
+          if (MarketingEmailErrors.is(err, 'SERVICE_UNAVAILABLE')) {
+            // the service is not available, ask to retry later
+            $('.spinner').hide();
+            $('.error').html(err).show();
+            setTimeout(function() {
+              self.navigate('/');
+            }, 2000);
+            console.log(self);
+          }
 
           // error fetching the email preferences, show the
           // error and hide the spinner.
-          self.$('.spinner').hide();
-          self.displayError(err);
+          self.$('.spinner').hide(); // <-- I don't think this ever works as intended
+          self.displayError(err); // <-- This logs to the console, but should actually display .error
           return false;
```

Comment:
self returns the proper object (a div with class communication-preferences as expected), but fails to hide the spinner or display the erro.